### PR TITLE
[WIP][caffe2] Extend caffe2 FC/FCTranspose op to handle 2d bias.

### PR DESCRIPTION
**Pull request by JerryShih**
Currently, FC/FCTranspose only accepts 1d bias and broadcasts it to 2d matrix. If we have 2d bias, we could just bypass it to FC/FCTranspose op.

### CODE REVIEW

**Comment by JerryShih** [comment]
@houseroad 
I haven't handle the gradient parts, but I would like to have the feedback for the forward path first.

File: caffe2/operators/fully_connected_op.h [hunk 1]
```diff
@@ -89,7 +95,22 @@ class FullyConnectedOp final : public Operator<Context> {
...
+    // copy bias to output y
+    if (b.ndim() == 2) {
+      const T_B* b_data = b.template data<T_B>();
+      T_Y *y_data = Y->template mutable_data<T_Y>();
+          math::CopyVector<T_B, Context>(M * N, b_data, y_data, &context_);
+    } else {
+      const T_B* b_data = b.template data<T_B>();
+      T_Y* y_data = Y->template mutable_data<T_Y>();
+      for (int i = 0; i < M; ++i) {
+        math::CopyVector<T_B, Context>(N, b_data, y_data, &context_);
```
Comment (JerryShih) [inline caffe2/operators/fully_connected_op.h]: In the original implementation, it uses gemm to create a m*n bias matrix from 1d bias vector. I just use CopyVector to do the broadcast. I haven't checked the performance difference with profiler, but there is no float multiplying in this way.

**Comment by ezyang** [comment]
@houseroad can you review this PR?

**Comment by JerryShih** [comment]
I will turn to add a gemm op in caffe2 instead of the fc op with 2d bias.

**Comment by houseroad** [comment]
Close the PR, since we have changed the design.

**Pull request closed without merging**